### PR TITLE
Add excluded_dirs support, add s3transfer excluded_dirs, fix #93 #108

### LINKED ISSUE #93: Hard-coded excluded_dirs
function.py hard-codes 8 directories that will always be excluded.  But this breaks deployment of any projects that happen to use those packages.  I discovered this when the "jmespath" dependency in my project mysteriously refused to be included in the deployed package.  Some code-spelunking revealed the hard-coded list in function.py.

Any chance of just removing that list entirely?  As far as I can tell kappa is not caching any of those dependencies in my src folder anyway, so I don't think it should need to exclude them.

### PR DESCRIPTION
**Pull request by stonemary**
What changed:

1. new feature: 
`excluded_dirs` are now configurable. #93 

**Before**: kappa in-explicitly exclude directories generated by some libraries. Those libraries are typically provided by AWS Lambda environment. Examples includes boto, boto3 etc. 
**Now**: excluded_dirs are now configurable in kappa.yml, user can specify `default`, `none`, or a list of their own customized directories to be excluded. 
**Backward compatibility**: the change is backward compatible, if `excluded_dirs` are not configured, it is in `default` and excludes folders generated by libraries provided by AWS Lambda - same as before. 

2. bug fix:
s3transfer is one of the folders/libraries boto3 installs, but it were not included in the excluded folders, which causes compatibility issues when AWS Lambda updates its buildin libraries under the hood. #108

### CODE REVIEW

File: kappa/function.py [hunk 1]
```diff
@@ -31,9 +31,9 @@
 
 class Function(object):
 
-    excluded_dirs = ['boto3', 'botocore', 'concurrent', 'dateutil',
-                     'docutils', 'futures', 'jmespath', 'python_dateutil']
-    excluded_files = ['.gitignore']
+    DEFAULT_EXCLUDED_DIRS = ['boto3', 'botocore', 's3transfer', 'concurrent', 'dateutil', 'docutils', 'futures',
```
Comment (josegonzalez) [inline kappa/function.py:34]: Why were these added?

Comment (stonemary) [reply kappa/function.py:34]: the only thing added is s3transfer. more details are in #108 , but in short:
> Kappa excludes folders like boto3 by default in its packaging process. However, s3transfer ... is installed during the installation of boto3 .. causes compatibility issues when user has a boto3 version different from the current Lambda buildin, and uploads a function with kappa.

Comment (stonemary) [reply kappa/function.py:34]: I added a summary on top. Hopefully it would help.

Comment (josegonzalez) [reply kappa/function.py:34]: Got it.
